network/p2p/streams.go

```diff
@@ -134,6 +136,19 @@ func (n *streamManager) Connected(net network.Network, conn network.Conn) {
 		return
 	}
 
+	// check if this is outgoing connection but made not by us (serviceImpl.dialNode)
+	// then it was made by some sub component like pubsub, ignore
+	if n.cfg.IsHybridServer() && conn.Stat().Direction == network.DirOutbound {
```

Comment:
Why do we not run this logic if P2P only node?

Reply:
This was requested by Chris and I agreed - in p2p mode we want more decentralized mesh and maybe relying on pubsub scoring to give us some extra good nodes is not a bad thing.